tests/Unit/Time/StepChoosers/Test_ErrorControl.cpp

```diff
@@ -0,0 +1,330 @@
+  CHECK(std::make_pair(previous_step, true) ==
+        error_control_base->desired_step(make_not_null(&box), previous_step,
+                                         cache));
+
+  db::mutate<Tags::StepperErrorUpdated>(
+      make_not_null(&box),
+      [](const gsl::not_null<bool*> stepper_updated) noexcept {
+        *stepper_updated = true;
+      });
+  const std::pair<double, bool> result = error_control(
+      previous_step_error,
+      db::get<Tags::HistoryEvolvedVariables<EvolvedVariablesTag>>(box), error,
+      true, time_stepper, previous_step, cache);
+  // reset the previous step error so we can reuse the former state on the next
+  // re-application
+  *previous_step_error = db::get<StepChoosers::Tags::PreviousStepError>(box);
+  CHECK(error_control_base->desired_step(make_not_null(&box), previous_step,
+                                         cache) == result);
+  db::mutate<StepChoosers::Tags::PreviousStepError>(
+      make_not_null(&box),
+      [previous_step_error](const gsl::not_null<std::optional<double>*>
+                                previous_step_error_from_box) noexcept {
+        *previous_step_error_from_box = *previous_step_error;
+      });
+  CHECK(serialize_and_deserialize(error_control)(
+            previous_step_error,
+            db::get<Tags::HistoryEvolvedVariables<EvolvedVariablesTag>>(box),
+            error, true, time_stepper, previous_step, cache) == result);
+  CHECK(serialize_and_deserialize(error_control_base)
+            ->desired_step(make_not_null(&box), previous_step, cache) ==
+        result);
+  return result;
+}
+
+// need to test both with two steps in the history and with one
+
+SPECTRE_TEST_CASE("Unit.Time.StepChoosers.ErrorControl", "[Unit][Time]") {
+  Parallel::register_derived_classes_with_charm<StepChooserType>();
+
+  MAKE_GENERATOR(generator);
+  std::uniform_real_distribution<> step_dist{1.0, 2.0};
+  Variables<tmpl::list<EvolvedVar1, EvolvedVar2>> step_values{5};
+  fill_with_random_values(make_not_null(&get<EvolvedVar1>(step_values)),
+                          make_not_null(&generator), make_not_null(&step_dist));
+  fill_with_random_values(make_not_null(&get<EvolvedVar2>(step_values)),
+                          make_not_null(&generator), make_not_null(&step_dist));
+  const DataVector flattened_step_values{step_values.data(), 15};
+  std::uniform_real_distribution<> error_dist{1.0e-4, 2.0e-4};
+  Variables<tmpl::list<Tags::StepperError<EvolvedVar1>,
+                       Tags::StepperError<EvolvedVar2>>>
+      step_errors{5};
+  fill_with_random_values(
+      make_not_null(&get<Tags::StepperError<EvolvedVar1>>(step_errors)),
+      make_not_null(&generator), make_not_null(&error_dist));
+  fill_with_random_values(
+      make_not_null(&get<Tags::StepperError<EvolvedVar2>>(step_errors)),
+      make_not_null(&generator), make_not_null(&error_dist));
+  const DataVector flattened_step_errors{step_errors.data(), 15};
+  const size_t stepper_order = 2;
```

Comment:
Sometimes you use this variable and sometimes a hard-coded 2, but in any case you should test another order.  (Comparing consistency between two orders looks not too bad, if working one out is annoying.)